CIP-0062/README.md

```diff
@@ -368,19 +347,28 @@ The [Signed Delegation Metadata](#signeddelegationmetadata) of the voter registr
 
 ### Examples of Message Flows and Processes
 
+#### Connection
+
+1. **dApp Dialogue** - User indicates to the dApp intention of wallet connection, the user selects their chosen wallet from a list of supported ones offered by the dApp.
+2. **Connection** - Using the name of the selected wallet the dApp can call [CIP-30's enable](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0030#cardanowalletnameenable-extensions-extension----promiseapi) passing in this CIPs identifier: `cardano.{walletName}.enable([{"cip": 62 }])`. This should cause the wallet to ask the user to grant permission for the dApp connecting and using the CIP-62 API Extension.
+3. **Purpose** - Once connection is established the dApp will want to establish the supported wallet's `VotingPurpose`s, this is achieved via `.getVotingPurposes()`. By knowing the supported purposes the dApp can correctly choose which ones to use when calling `.signVotes()` and `.submitDelegation()`.
+
 #### Delegation
 
 Recall from [CIP-36](https://github.com/cardano-foundation/CIPs/tree/master/CIP-0036) a registration is a self-delegation, allocating one's voting power to one's own voting key.
 
 1. **Get Voting Key** - dApp calls the method `api.getVotingCredentials()` to return the connected wallet account's public `voteKey`.
```

Comment:
not related to this PR, but I was just polishing HW wallet support when I realized that only Ledger HW wallet will be able to return the voting key. So this call would fail for a Trezor user even though they may want just to delegate their voting rights and for that usecase it's sufficient to get the staking key. One solution I can think of is to split this call into two, so each piece of information is available independently, i.e. `getVotingKey` and `getStakingKey` - what do you think @Ryun1 ?

Reply:
@stevenj

Reply:
for the record - it seems that the `stakingKeyPub` may not need to be exposed at all as the wallet can handle it internally. For the sake of constructing the delegation transaction, it's enough for the wallet to receive the staking address in the `DelegatedCertficate` [structure](https://github.com/ehanoc/CIPs/blob/39d575ca0f7604d7be2d4d5a9ab28d750b39006e/CIP-0062/README.md?plain=1#L151) and it can map it back to the respective public key without issues.

If we suppose that a wallet has a single staking key, not even the staking address would be needed, but it seems to be more future proof to include it as the CIP-30 API itself exposes the reward (i.e. staking) addresses as an array: https://github.com/cardano-foundation/CIPs/blob/master/CIP-0030/README.md?plain=1#L305

So to sum it up - unless there's some usecase I'm missingI think the `stakingKeyPub` can be dropped from the `getVotingCredentials()` call altogether in which case it probably makes sense to rename the call back to `getVotingKey()` and the `submitDelegation` call could reference just the `stakingAddress` which can be retrieved from the CIP-30 `getRewardAddresses()` call. This would also solve the concern raised in the question above